modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExchangeServiceImpl.java

```diff
@@ -112,45 +109,28 @@ public CompletableFuture<Void> request(String nodeName, UUID queryId, long fragm
     /** {@inheritDoc} */
     @Override
     public CompletableFuture<Void> sendError(String nodeName, UUID queryId, long fragmentId, Throwable error) {
-        IgniteException errorWithCode = wrapIfNecessary(error);
+        Throwable errorWithCode = IgniteExceptionMapperUtil.mapToPublicException(ExceptionUtils.unwrapCause(error));
```

Comment:
Why do we convert exception here?
Seems, we can convert internal Traceble exception to default one with INTERNAL_ERR code and get lost traceID.

Reply:
Thanks, seems there is no need to convert exception here. fixed.